packages/next/src/server/web/spec-extension/adapters/request-cookies.ts

```diff
@@ -144,29 +152,83 @@ export class MutableRequestCookiesAdapter {
               )
               try {
                 target.delete(...args)
+                return wrappedCookies
               } finally {
                 updateResponseCookies()
               }
             }
           case 'set':
-            return function (
-              ...args:
-                | [key: string, value: string, cookie?: Partial<ResponseCookie>]
-                | [options: ResponseCookie]
-            ) {
+            return function (...args: SetCookieArgs) {
               modifiedCookies.add(
                 typeof args[0] === 'string' ? args[0] : args[0].name
               )
               try {
-                return target.set(...args)
+                target.set(...args)
+                return wrappedCookies
```

Comment:
and another small bugfix: if someone did `cookieStore.set(...).set(...)`, the second `set` would be called on `target`, i.e. the underlying ResponseCookies object, **not the proxy**, so we'd miss an `updateResponseCookies` call